src/install/install.zig

```diff
@@ -9512,18 +9500,21 @@ pub const PackageManager = struct {
 
             installer.completeRemainingScripts(log_level);
 
-            if (root_lifecycle_scripts_count > 0) {
-                // root lifecycle scripts can run now that all dependencies are installed
-                // and their lifecycle script have finished
-                try this.spawnPackageLifecycleScripts(ctx, this.root_lifecycle_scripts.?, log_level);
+            if (!FeatureFlags.breaking_changes_1_1_0) {
+                if (this.root_lifecycle_scripts) |scripts| {
+                    if (comptime log_level.showProgress()) {
+                        scripts_node.setEstimatedTotalItems(scripts_node.unprotected_completed_items + scripts.total);
+                    }
+                    try this.spawnPackageLifecycleScripts(ctx, scripts, log_level, false);
+                }
             }
 
             while (this.pending_lifecycle_script_tasks.load(.Monotonic) > 0) {
                 if (PackageManager.verbose_install) {
                     if (PackageManager.hasEnoughTimePassedBetweenWaitingMessages()) Output.prettyErrorln("<d>[PackageManager]<r> waiting for {d} scripts\n", .{this.pending_lifecycle_script_tasks.load(.Monotonic)});
                 }
 
-                PackageManager.instance.sleep();
+                this.sleep();
```

Comment:
What does sleeping allow?

Reply:
While we wait for lifecycle scripts to finish, sleeping prevents us from using 100% of the cpu